packages/react-components/react-dialog/Spec.md

```diff
@@ -1,63 +1,459 @@
+          <DialogTitle>This is an alert</DialogTitle>
+          <DialogBody>This is going to be inside the dialog</DialogBody>
+          <DialogActions>
+            <Button onClick={handleClose}>Close</Button>
+            <Button>Action</Button>
+          </DialogActions>
+        </DialogContent>
+      </Dialog>
+    </>
+  );
+};
+```
+
+```html
+<button aria-haspopup="true" class="fui-button">Open Dialog</button>
+<!-- ... portal ... -->
+<div aria-hidden="true" class="fui-dialog-overlay"></div>
+<div
+  aria-describedby="fui-dialog-body-id"
+  aria-labelledby="fui-dialog-title-id"
+  aria-modal="true"
+  role="alertdialog"
+  class="fui-dialog-content"
+>
+  <div id="fui-dialog-title-id" class="fui-dialog-title">
+    <span>Title</span>
+    <!-- closeButton -->
+  </div>
+  <div id="fui-dialog-body-id" class="fui-dialog-body">This is going to be inside the dialog</div>
+  <div class="fui-dialog-actions">
+    <button class="fui-button">Close</button>
+    <button class="fui-button">Action</button>
+  </div>
+</div>
+<!-- ... portal ... -->
+```
+
+## Migration
+
+_TBA: Link to migration guide doc_
 
 ## Behaviors
 
-_Explain how the component will behave in use, including:_
+Dialog will use **Tabster** to handle the keyboard navigation and ensure focus trapping.
+
+The below references were used to decide appropriate keyboard interaction from an a11y perspective.
+
+- https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/
+- https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog.html
+- https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/alertdialog.html
+
+### Modal
+
+#### Mouse & Touch
+
+![Mouse and touch behavior of Modal dialog](./assets/ModalDialogMouseInteraction.png)
+
+1. Clicking on the trigger (element / button component) a Dialog is displayed with a dimmed background.
+2. Clicking on the dimmed background dismisses the dialog.
+3. The dialog can be dismissed also when confirmation button is clicked (footer).
+
+#### Keyboard
+
+![Keyboard behavior of a modal dialog](./assets/ModalDialogKeyboardInteraction.png)
+
+1. **(1)** TabKey to set focus on Trigger, use EnterKey to open.
+2. **(2-6)** Focus is moved to the first focusable control inside the dialog.
+3. **(5-6)** After the dialog is dismissed, keyboard focus should be moved back to where it was before it moved into the dialog. Otherwise the focus can be dropped to the beginning of the page. Or if the item is no longer available it can be moved to the next logical location in that region i.e. next / previous item.
+4. **TabKey** Moves focus to next focusable element inside the dialog. When focus is on the last focusable element in the dialog, moves focus to the next focusable action in the browser window.
+5. **Shift+Tab** Moves focus to previous focusable element inside the dialog. When focus is on the first focusable element in the dialog, moves focus to the last focusable action within the browser window.
+6. **EscKey** Closes the dialog.
```

Comment:
it would be good to mention, in which cases the component can return focus to the trigger - and when the consumer would be responsible for that